frontend/src/app/services/auth/auth.service.ts

```diff
@@ -0,0 +1,64 @@
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { JwtHelperService } from '@auth0/angular-jwt';
+import { BehaviorSubject, catchError, Observable, tap, throwError } from 'rxjs';
+import { DecodedToken } from 'src/app/interfaces/decoded-token';
+import { environment } from 'src/environments/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  public api: string = environment.apiUrl;
+  private token: string = null;
+  private jwtHelper = new JwtHelperService();
+  public isUserLoggedIn: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+
+  constructor(
+    private http: HttpClient, 
+    ) { }
+
+  public login(email: string, password: string): Observable <{token: string}> {
+    return this.http.post<{token: string}>(`${this.api}/authenticate`, {email: email, password: password})
+      .pipe(
+        tap(
+          ({token}) => {
+            localStorage.setItem('auth_token', token);
+            this.setToken(token);
+            this.isUserLoggedIn.next(true);
+          }
+        ),
+        catchError((error: HttpErrorResponse) =>  throwError(error.error.errorMessage))
```

Comment:
Will be better to handle errors in component where you are using this login method